### PR TITLE
feat: sdk7 debug panel scene metrics

### PR DESCRIPTION
**Pull request by pravusjif**
Debug panel scene metrics were not implementet, now it's hooked up to the same datastore logic that SDK6 uses.

Issue: https://github.com/decentraland/sdk/issues/1032

-------------------------

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at a219ba6</samp>

This pull request adds a debug mode feature to the GltfContainer, Material, and MeshRenderer components. The debug mode allows these components to update the scene metrics data store with information about the loaded and unloaded assets. The pull request also refactors some classes to use fields instead of parameters and fixes some typos and inconsistencies. The pull request modifies several assembly definition files to add the necessary references to the DataStore and DebugConfig namespaces. The pull request also updates the tests to pass the debug config object to the constructors of the component handlers.

-------------------------

### QA TEST INSTRUCTIONS

1. Enter https://sdk-test-scenes.decentraland.zone/?NETWORK=mainnet&position=1%2C-2&SCENE_DEBUG_PANEL=&realm=LocalPreview&explorer-branch=feat/sdk7-debug-panel-scene-metrics
2. Open the "SCENE" debug panel metrics

Screen Shot 2023-11-27 at 12 52 06

3. Confirm that every time you spawn a new cube, the amount of Poly and Material increases

Screen Shot 2023-11-27 at 12 52 41

### CODE REVIEW

File: unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/GltfContainer/GltfContainerHandler.cs [hunk 1]
```diff
@@ -464,5 +437,31 @@ void RemoveActiveColliders(IList<Collider> colliders, IInternalECSComponent<Inte
...
+
+        private void AddCurrentRendereableToSceneMetrics()
+        {
+            DataStore.i.sceneWorldObjects.AddRendereable(entity.scene.sceneData.sceneNumber, currentRendereable);
+
+            foreach (Renderer renderer in renderers)
+            {
+                foreach (Material material in renderer.sharedMaterials)
+                {
+                    DataStore.i.sceneWorldObjects.AddMaterial(entity.scene.sceneData.sceneNumber, entity.entityId, material);
```
Comment (Kinerius) [inline unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/GltfContainer/GltfContainerHandler.cs:449]: This class has already some data stores being injected, can we do the same with this one?

File: unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Material/MaterialHandler.cs [hunk 2]
```diff
@@ -34,7 +45,12 @@ public void OnComponentRemoved(IParcelScene scene, IDCLEntity entity)
 
             while (activePromises.Count > 0)
             {
-                AssetPromiseKeeper_Material.i.Forget(activePromises.Dequeue());
+                var promise = activePromises.Dequeue();
+
+                if (isDebugMode)
+                    DataStore.i.sceneWorldObjects.RemoveMaterial(scene.sceneData.sceneNumber, entity.entityId, promise.asset.material);
```
Comment (Kinerius) [inline unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Material/MaterialHandler.cs:51]: Same here,  `DataStore.i.sceneWorldObjects` can be injected

**Review by Kinerius — APPROVED** [review]
PR Looks ok, I left some comments for correctness but anyways I'm going to approve 👍

File: unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/GltfContainer/GltfContainerHandler.cs (hunk 1 above)
Comment (pravusjif) [reply unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/GltfContainer/GltfContainerHandler.cs]: Sounds good, done!

File: unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Material/MaterialHandler.cs (hunk 2 above)
Comment (pravusjif) [reply unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Material/MaterialHandler.cs]: Done!

**Pull request merged**